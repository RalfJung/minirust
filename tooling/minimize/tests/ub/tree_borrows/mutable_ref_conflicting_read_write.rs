//@ compile-flags: --minimize-tree-borrows
fn main() {
    let x = &mut 31; // (x, Reserved)   
    let xraw = x as *mut i32; // (x, Reserved) 
    let y = unsafe { &mut *xraw }; // (x, Reserved) -> (y, Reserved)
    *y = 42; // (x, Active) -> (y, Active)
    assert!(unsafe { *xraw } == 42); // (x, Active) -> (y, Frozen)
    *y = 31; // UB! Child Write to Frozen 
}
