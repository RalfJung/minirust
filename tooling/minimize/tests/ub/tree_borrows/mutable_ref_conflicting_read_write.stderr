fatal error: UB: Tree Borrows: Child writing a pointer with the Frozen permission
